mptcp-info.md

```diff
@@ -0,0 +1,379 @@
+---
+layout: home
+title: MPTCP info (Devs)
+nav_order: 5
+nav_titles: true
+titles_max_depth: 2
+---
+
+This section is for **application developers** willing to retrieve MPTCP
+specific info from their app on Linux. End-users can retrieve info via other
+interfaces, see the [Debugging](debugging.html) section.
+
+Like TCP, it is possible to retrieve MPTCP info from the socket, via
+[`getsockopt()`](https://www.man7.org/linux/man-pages/man2/getsockopt.2.html),
+using `MPTCP_INFO` (`1`). It is also possible to retrieve info from the subflows
+via `MPTCP_TCPINFO` (`2`), `MPTCP_SUBFLOW_ADDRS` (`3`), and `MPTCP_FULL_INFO`
+(`4`).
+
+## MPTCP socket level
+
+`MPTCP_INFO` is the easiest way to retrieve info at the MPTCP level. Note that
+these stats are also exposed via the Netlink Diag interface, and can be seen
+with `ss -Mi` for example.
+
+To retrieve info, `getsockopt(MPTCP_INFO)` can be used: the kernel will fill a
+given buffer of a given size with items following the `mptcp_info` structure,
+and update the variable with the given size with the number of bytes that have
+been written:
+
+<div class="language-c highlighter-rouge">
+  <div class="highlight">
+    <pre class="highlight"><code>int err = <span class="color-main">getsockopt</span>(<span class="color-blue">mptcp_fd</span>, <span class="color-yellow">SOL_MPTCP</span>, <span class="color-green">MPTCP_INFO</span>, &<span class="color-red">buffer</span>, &<span class="color-orange">optlen</span>)</code></pre>
+  </div>
+</div>
+
+{: .warning}
+It is important to note that this `mptcp_info` structure got new items amongst
+versions, and might continue to grow. It means that the userspace application
+and the kernel might work with different versions of this structure. The
+kernel will write the amount of bytes that have been written in the
+`optlen`{: .color-orange} variable, so the userspace application can detect what
+the current kernel has exposed.
+
+The kernel might write fewer data than what the userspace application expects.
```

Comment:
Funny, I initially wrote `less`, but then switched to `fewer` because of https://languagetool.org/insights/post/grammar-fewer-less/#when-to-use-%E2%80%9Cfewer%E2%80%9D
(`data` is "uncountable", but I think it is no longer used like that today).

I prefer `less` if it sounds better!